pom.xml

```diff
@@ -162,7 +179,7 @@
     </dependency>
     <dependency>
       <groupId>co.cask.cdap</groupId>
-      <artifactId>cdap-etl-api-spark</artifactId>
+      <artifactId>cdap-etl-api-spark2_2.11</artifactId>
```

Comment:
Instead of this, can you try excluding spark-core and spark-streaming from the cdap-etl-api-spark dependency and see if that works.

Reply:
Solution with exclusions is possible but makes it harder to write new plugins and requires from plugin developer a knowledge about cdap dependency structure. Isn't the main goal of template projects to make life easier for plugin developers?

Reply:
actually, I think you don't need to exclude anything at all... just having spark2 in the dependencies should take precedence over a transitive dependency. Were you having issues with cdap-etl-api-spark and spark-core2?

Reply:
I have just checked and it works fine. At first I didn't realize spark template module has only provided dependencies. I was just trying to follow convention from other spark 2 modules. If it is not needed and it's fine to keep dependency set for plugin like it was before, then I should close https://github.com/cdapio/cdap/pull/11023 and move back to cdap-etl-api-spark dependency. 
By the way, what about archetypes? Shouldn't it be updated for compatibility with Spark 2 in the future?